src/dev/i18n/extract_html_messages.js

```diff
@@ -0,0 +1,142 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { jsdom } from 'jsdom';
+import { parse } from '@babel/parser';
+import { isDirectiveLiteral, isObjectExpression, isStringLiteral } from '@babel/types';
+
+import { isPropertyWithKey, escapeLineBreak, traverseNodes } from './utils';
+import { DEFAULT_MESSAGE_KEY, CONTEXT_KEY } from './constants';
+
+/**
+ * Find all substrings of "{{ any text }}" pattern
+ */
+const ANGULAR_EXPRESSION_REGEX = /\{\{+([\s\S]*?)\}\}+/g;
+
+const I18N_FILTER_MARKER = '| i18n: ';
+
+/**
+ * Extract default message from an angular filter expression argument
+ * @param {string} expression JavaScript code containing a filter object
+ * @returns {string} Default message
+ */
+function parseFilterObjectExpression(expression) {
+  // parse an object expression instead of block statement
+  const nodes = parse(`+${expression}`).program.body;
+
+  for (const node of traverseNodes(nodes)) {
+    if (!isObjectExpression(node)) {
+      continue;
+    }
+
+    let message;
+    let context;
+
+    for (const property of node.properties) {
+      if (isPropertyWithKey(property, DEFAULT_MESSAGE_KEY)) {
+        if (!isStringLiteral(property.value)) {
+          throw new Error('defaultMessage value should be a string literal.');
+        }
+
+        message = escapeLineBreak(property.value.value);
+      } else if (isPropertyWithKey(property, CONTEXT_KEY)) {
+        if (!isStringLiteral(property.value)) {
+          throw new Error('context value should be a string literal.');
+        }
+
+        context = property.value.value;
+      }
+    }
+
+    return { message, context };
+  }
+}
```

Comment:
__nit:__ we should probably return `null` here if we couldn't find any node with message just to explicitly show that it may be the case.